File: solutions/04.cs
public class Four
{
	static void Main()
	{
		int a = 0;
		int b = -1;
		int c = 999;

		for (int i = 100; i <= c; i++)
			for (int j = 100; j <= c; j++)
			{
				a = i * j;

				if ((a == Reverse(a)) && (a > b))
					b = a;
			}

		Console.WriteLine("\n{0}\n", b);
	}

	static int Reverse(int m)
	{
		int n = 0;

		while (m != 0)
		{
			n *= 10;
			n += m % 10;
			m /= 10;
		}

		return m;
	}
}

File: solutions/ruby/04.rb
#!/usr/bin/env ruby

def palindrome?(str)
    str == str.reverse
end

def largest_palindrome(range)
    palindromes = []

    range.each { |x|
        range.each { |y|
            xy = x * y
            palindromes << xy if palindrome?(xy.to_s)
        }
    }

    palindromes.max
end

p largest_palindrome 100..999
